apps/ledger-live-mobile/src/newArch/features/Web3Hub/__integrations__/web3hub.integration.test.tsx

```diff
@@ -316,4 +322,50 @@ describe("Web3Hub integration test", () => {
     await user.press(screen.getByRole("button", { name: /back/i }));
     expect(await screen.findByText("Explore web3")).toBeOnTheScreen();
   });
+
+  /** @TODO https://github.com/LedgerHQ/ledger-live/pull/8050 : unskip after adding state manager to the DB */
+  it.skip("Should list manifests, add to recently used, go to search and display them", async () => {
+    const { user } = render(<Web3HubTest />);
+
+    expect(await screen.findByText("Explore web3")).toBeOnTheScreen();
+
+    await waitForLoader();
+
+    expect(await screen.findByRole("searchbox")).toBeOnTheScreen();
+    expect(screen.getByRole("searchbox")).toBeDisabled();
+    await user.press(screen.getByRole("searchbox"));
+    expect(await screen.findByRole("searchbox")).toBeOnTheScreen();
+    expect(screen.getByRole("searchbox")).toBeEnabled();
+
+    expect(screen.queryByText("Recently used")).toBeNull();
+
+    expect(await screen.findByRole("button", { name: /back/i })).toBeOnTheScreen();
+    await user.press(screen.getByRole("button", { name: /back/i }));
+    expect(await screen.findByText("Explore web3")).toBeOnTheScreen();
+
+    expect((await screen.findAllByText("Dummy Wallet App"))[0]).toBeOnTheScreen();
+    await user.press(screen.getAllByText("Dummy Wallet App")[0]);
+    expect(await screen.findByText("Do not remind me again.")).toBeOnTheScreen();
+    expect(await screen.findByText("Open Dummy Wallet App")).toBeOnTheScreen();
+    await user.press(screen.getByText("Open Dummy Wallet App"));
+    expect(await screen.findByText("dummy-0")).toBeOnTheScreen();
+    expect(await screen.findByText("Dummy Wallet App")).toBeOnTheScreen();
+
+    expect(await screen.findByRole("button", { name: /back/i })).toBeOnTheScreen();
+    await user.press(screen.getByRole("button", { name: /back/i }));
+    expect(await screen.findByText("Explore web3")).toBeOnTheScreen();
+
+    expect(await screen.findByRole("searchbox")).toBeOnTheScreen();
+    expect(screen.getByRole("searchbox")).toBeDisabled();
+    await user.press(screen.getByRole("searchbox"));
+    expect(await screen.findByRole("searchbox")).toBeOnTheScreen();
+    expect(screen.getByRole("searchbox")).toBeEnabled();
+
+    expect(await screen.findByText("Recently used")).toBeOnTheScreen();
+    expect(await screen.findByTestId("recently-used-dummy-0")).toBeOnTheScreen();
```

Comment:
We also want to limit the usage of testIds to the bare minimum and try to use other user friendly way to find elements
https://testing-library.com/docs/queries/about#priority